plugins/bufferlines/barbecue.nix

```
@@ -0,0 +1,172 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.barbecue;
+  helpers = import ../helpers.nix {inherit lib;};
+  mkListStr = helpers.defaultNullOpts.mkNullable (types.listOf types.str);
+in {
+  options.plugins.barbecue =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "barbecue-nvim";
+
+      package = helpers.mkPackageOption "barbecue-nvim" pkgs.vimPlugins.barbecue-nvim;
+
+      attachNavic = helpers.defaultNullOpts.mkBool true ''
+        Whether to attach navic to language servers automatically.
+      '';
+
+      createAutocmd = helpers.defaultNullOpts.mkBool true ''
+        Whether to create winbar updater autocmd.
+      '';
+
+      includeBuftypes = mkListStr ''[ "" ]'' ''
+        Buftypes to enable winbar in.
+      '';
+
+      excludeFiletypes = mkListStr ''[ "netrw" "toggleterm" ]'' ''
+        Filetypes not to enable winbar in.
+      '';
+
+      modifiers = {
+        dirname = helpers.defaultNullOpts.mkStr ":~:." ''
+          Filename modifiers applied to dirname.
+
+          See: `:help filename-modifiers`
+        '';
+
+        basename = helpers.defaultNullOpts.mkStr "" ''
+          Filename modifiers applied to basename.
+
+          See: `:help filename-modifiers`
+        '';
+      };
+
+      showDirname = helpers.defaultNullOpts.mkBool true ''
+        Whether to display path to file.
+      '';
+
+      showBasename = helpers.defaultNullOpts.mkBool true ''
+        Whether to display file name.
+      '';
+
+      showModified = helpers.defaultNullOpts.mkBool false ''
+        Whether to replace file icon with the modified symbol when buffer is modified.
+      '';
+
+      modified = helpers.defaultNullOpts.mkStr ''function(bufnr) return vim.bo[bufnr].modified end,'' ''
+        Get modified status of file.
+
+        NOTE: This can be used to get file modified status from SCM (e.g. git)
+      '';
+
+      showNavic = helpers.defaultNullOpts.mkBool true ''
+        Whether to show/use navic in the winbar.
+      '';
+
+      leadCustomSection = helpers.defaultNullOpts.mkStr ''function() return " " end,'' ''
+            Get leading custom section contents.
+
+        NOTE: This function shouldn't do any expensive actions as it is run on each render.
+      '';
```

```suggestion
      leadCustomSection =
        helpers.defaultNullOpts.mkStr
        ''
          function()
            return " "
          end
        ''
        ''
          Get leading custom section contents.
          NOTE: This function shouldn't do any expensive actions as it is run on each render.
        '';
```
Prefer this formatting for those options.

---

yeh looks much better, will update the others.